packages/gestalt/src/Upsell.js

```diff
@@ -0,0 +1,164 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Box from './Box.js';
+import Heading from './Heading.js';
+import IconButton from './IconButton.js';
+import Button from './Button.js';
+import Text from './Text.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+import styles from './Upsell.css';
+
+type LinkData = {|
+  accessibilityLabel?: string,
+  href: string,
+  label: string,
+  onClick?: AbstractEventHandler<
+    | SyntheticMouseEvent<HTMLButtonElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLButtonElement>
+  >,
+|};
+
+type Props = {|
+  dismissButton?: {|
+    accessibilityLabel: string,
+    onDismiss: () => void,
+  |},
+  message: string,
+  primaryLink?: LinkData,
+  secondaryLink?: LinkData,
+  title?: string,
+|};
+
+const UpsellLink = ({
+  data,
+  stacked,
+  type,
+}: {|
+  data: LinkData,
+  stacked?: boolean,
+  type: string,
+|}): Node => {
+  const color = type === 'primary' ? 'red' : 'gray';
+  const { accessibilityLabel, href, label, onClick } = data;
+
+  return (
+    <Box
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+      paddingX={1}
+      marginTop={type === 'primary' && stacked ? 2 : undefined}
+      mdMarginTop="auto"
+      mdMarginBottom="auto"
+      rounding={type === 'primary' ? 'pill' : undefined}
+    >
+      <Button
+        accessibilityLabel={accessibilityLabel}
+        color={color}
+        href={href}
+        onClick={onClick}
+        role="link"
+        size="lg"
+        text={label}
+      />
+    </Box>
+  );
+};
+
+export default function Upsell({
+  dismissButton,
+  message,
+  primaryLink,
+  secondaryLink,
+  title,
+}: Props): Node {
+  return (
+    <Box
+      display="flex"
+      direction="column"
+      mdDirection="row"
+      padding={6}
+      mdPadding={8}
+      position="relative"
+      rounding={4}
+      borderStyle="shadow"
+    >
+      <Box
+        mdDisplay="flex"
+        wrap
+        width="100%"
+        mdMarginTop={-3}
+        mdMarginBottom={-3}
+      >
+        <Box
+          display="flex"
+          marginBottom={primaryLink || secondaryLink ? 4 : undefined}
+          mdMarginBottom={primaryLink || secondaryLink ? 0 : undefined}
+          mdPaddingY={3}
+        >
+          <Box marginBottom="auto" marginTop="auto" maxWidth={648} paddingX={6}>
+            {title && (
+              <Box marginBottom={2}>
+                <Heading size="sm">{title}</Heading>
+              </Box>
+            )}
+            <Text>{message}</Text>
+          </Box>
+        </Box>
+        <Box mdDisplay="flex" marginStart="auto" mdPaddingY={3}>
+          {secondaryLink && (
+            <UpsellLink type="secondary" data={secondaryLink} />
+          )}
+          {primaryLink && (
```

Comment:
One thing to note is that in the design spec for Upsell (and Callout) there is 48px of margin on the right side of this button. This is _not_ consistent with the current implementation of Callout where there is only 32px on margin on the right side. I opted to be consistent with the current implementation of Callout, but let me know if I should make this 48px, make both 48px or leave it as is.

Reply:
Correct. Both Upsell and Callout have a 48px padding on the right while a 32px padding on the left. 
I don't know if it's the finest solution but this fixes it/
```
        <Box mdDisplay="flex" marginStart="auto" mdMarginEnd={4}>
          {secondaryLink && (
            <CalloutLink type="secondary" data={secondaryLink} />
          )}
          {primaryLink && (
            <CalloutLink
              stacked={!!secondaryLink}
              type="primary"
              data={primaryLink}
            />
          )}

```

Reply:
Updated here and Callout! Thank you!!

<img width="818" alt="Screen Shot 2020-11-05 at 3 13 37 PM" src="https://user-images.githubusercontent.com/46180742/98306880-881c6c00-1f79-11eb-8379-8757a8d807bc.png">
<img width="825" alt="Screen Shot 2020-11-05 at 3 13 25 PM" src="https://user-images.githubusercontent.com/46180742/98306883-894d9900-1f79-11eb-9f76-8f20cdd90399.png">